### DISCUSSION TITLE
Add-on for medial player and image viewer

### DISCUSSION THREAD

**Discussion by Jpersh27** [TeamAmaze/AmazeFileManager#2982] [General]
Is your feature request related to a problem? 

• Yes

A clear and concise description of what the problem is.

• The file manager lacks an internal Media Player and at this point In development it desperately needs one. 

• It is frustrating and annoying to have to use an external application to play video files each and every time. It slows me down and requires me to use two applications at once instead of being able to view all my files inside one application.

**Describe the solution you'd like**

• A built-in Media player that recognizes and plays all current video file types and codecs that are widely used. 

**Describe alternatives you've considered**

• No alternative options to consider or have considered that would be similar.

**Additional context**

• If the concern is that you would like to implement an open source Media player to the application and have not found a suitable one may I suggest the following below as an option.

• https://github.com/google/ExoPlayer

• @VishalNehra @arpitkh96 @EmmanuelMess  and @TranceLove

**Comment by VishalNehra** [comment]
Don't worry I'm already working. You can expect it soon

**Comment by Jpersh27** [comment]
@VishalNehra  Thank you for the reply and the update.

• I however do have a couple follow up questions I would like to ask you that I have listed here down below.

1.) Do you have an estimated or exact ETA on when you expect you'll be able to release an update for the application with the internal Media Player added? 

2.) Do you have any screenshots you can share showing how the Media Player will function within the application?

3.) Will it be native to the application or require a separate add-on?

Thank and I will be looking forward to your reply soon.

**Comment by EmmanuelMess** [comment]
A media player is not something that should be in a File Manager, expect it to be an extension.

**Comment by VishalNehra** [comment]
Yes, it'll be a separate add-on. I can't give an ETA yet. Maybe few weeks.

**Comment by Jpersh27** [comment]
@VishalNehra it has been a few weeks and still no news or ETA on when this will be added as an external add-on???

**Comment by VishalNehra** [comment]
@Jpersh27 I've been caught up with stuff. Got absolutely no time. But I am bit free in upcoming 2 weeks. Let's hope I can finish in this time.

**Comment by Jpersh27** [comment]
@VishalNehra Thank you for the update I appreciate it and will be looking forward to hearing back from you soon.

@VishalNehra Can you also please consider reopening this case? https://github.com/TeamAmaze/AmazeFileManager/issues/2898 

• I feel as though that it was unnecessary for @EmmanuelMess to close it in the first place and feel like it would be a good way to track the progress of this issue/Add-on until it's released.

**Comment by VishalNehra** [comment]
I have migrated this to discussion. Will keep the status updated here.

**Reply by Jpersh27** [reply]
@VishalNehra Sounds good to me and thank you!

**Reply by Jpersh27** [reply]
@VishalNehra Any updates on when or if this feature will ever be implemented in the future?

**Reply by Jpersh27** [reply]
@VishalNehra ????

**Reply by VishalNehra** [reply]
Image viewer and video player is done. Currently working on music player. 
I'll publish on play Store for early testing once i have something concrete.

**Reply by Jpersh27** [reply]
@VishalNehra Any idea or ETA of when you plan on releasing it for testing either on GitHub or Google Play?

**Reply by Jpersh27** [reply]
@VishalNehra ????

**Reply by VishalNehra** [reply]
My target is 19th December.

**Reply by Jpersh27** [reply]
@VishalNehra Thank you for the update.

**Reply by Jpersh27** [reply]
@VishalNehra Will you be able to meet your deadline of December 19th or will it be pushed back????

**Reply by VishalNehra** [reply]
Yes, it won't be production ready though, there may be quite a lot of bugs. On top of that, it's just video player, music player and image viewer for now. I plan on adding a pdf / epub viewer, and if possible, a doc viewer as well.
So still a long way to go for production, but i'm making it as quickly as possible.

**Reply by Jpersh27** [reply]
@VishalNehra If there is no chance it releases by the end of the day today, When will the new deadline be pushed to now?

**Reply by VishalNehra** [reply]
Ok. I've added most of the things above. We now have a image viewer, video player, music player, doc viewer, pdf viewer, epub viewer.
I've contacted someone to make icons and banners (these are needed before app is published to play store). Meanwhile, I've decided to include a storage cleanup manage and file transfer feature in the app as well, much like files app by google. 
So basically this'll be a standalone app now instead of just a plugin.
That said, it'll take 3-4 more weeks before all these things are done (and i iron out some bugs). Hopefully by 16th Jan if all goes well, we'll have a really good utilities app.

**Reply by Jpersh27** [reply]
@VishalNehra Can you provide any screenshots or a video displaying these specific features in use? And thank you again @VishalNehra for the update!

**Reply by VishalNehra** [reply]
I'll tell you what, I'll publish pre release builds on play Store starting this weekend. Production release is still targeted for Jan, I'll be pushing updates and bug fixes till then. Because i don't see point in giving screenshot or videos. You can also help with testing

**Reply by Jpersh27** [reply]
@VishalNehra Thank you and sounds like a good idea. Looking forward to trying it out.

**Reply by VishalNehra** [reply]
@Jpersh27 Seems like a delay in ironing things out. I'll take till this weekend to publish initial version. 
I know is annoying but i'm trying my best here.

**Reply by Jpersh27** [reply]
@VishalNehra Thank you for the update I do appreciate that. If I may ask, what day are you planning for this weekend for the initial pre release build? And once you do release it can you provide an update here in this discussion thread?  And provide a direct link to the play store to download the application?

**Reply by VishalNehra** [reply]
Please join our telegram group, I'll be posting updates there starting tonight at https://t.me/AmazeFileManager

**Reply by Jpersh27** [reply]
@VishalNehra Thanks but No thanks. I would rather not use telegram because it is not a secure or safe platform to use and filled with nothing but malicious content and I would use a discord server to engage and talk to other users regarding the application. 

•Telegram isn't very great for direct feedback of users because only Administrators of the specific Telegram channel can send direct messages to users, all messages are decrypted on telegram servers and are only encrypted when they are sent on the users side, Only way for dual encryption is by turning on secret mode, No 2FA besides using your phone #, etc... I can go on and on....

1.) Why are you no longer going to provide updates via this discussion like you initially said? 

(It seems as though now your migrating everything over to that specific telegram channel). 

2.) Will the release be today or tomorrow?

**Reply by VishalNehra** [reply]
It's not like I'm migrating everything to telegram channel 'now'. These are prealpha builds, like i said earlier, instead of providing a video I'd rather have you try it and give feedback. So for this I feel the best medium is our telegram channel.
We already have over 800 users on this channel and we discuss development there. On top of that, we have our bot working which gives regular apks automatically by syncing with Github. The error reporting is also handled by bot there. 
It'll be too cumbersome to provide updates here, as I'm changing things in app quite frequently right now, so giving apk file for each change will be very daunting task for me here, but bot will generate and provide apk for each change automatically on telegram.
Also i would like to take help from existing user base on telegram for feedback and testing.

**Reply by shuvashish76** [reply]
> I would rather not use telegram because it is not a secure or safe platform to use and filled with nothing but malicious content and I would use a discord server to engage and talk to other users regarding the application.

Discord collects Aggregated Information (customer demographics, interests and behavior), other personal information like behavior of other service connected through Discord etc. They can share our information with 3rd party companies, may disclose your information if required etc. Source: https://discord.com/privacy
Then how Discord is different than other social platforms?
I respect your preffered social platform but I don't see a single better point for Discord. (though it has `server>multiple channel` feature which is something but not important for Amaze)

> filled with nothing but malicious content

Depends on your view point.

> only Administrators of the specific Telegram channel can send direct messages to users.

[@AmazeFileManager](https://t.me/AmazeFileManager), its group not channel.

> all messages are decrypted on telegram servers and are only encrypted when they are sent on the users side, Only way for dual encryption is by turning on secret mode, No 2FA besides using your phone.

Everything has some -ve points. Privacy depends on your own level of threat model & it varies person to person.
WhatsApp messages are end2end encrypted, does that mean it's privacy friendly?

I'm not a fan of Telegram in fact I don't believe half of the claims posted by Pavel Durov on his [channel](https://t.me/durov).
@VishalNehra sorry for going little bit offtopic.

---

Now why Telegram (from Amaze & general users' prospective)
- it offers unlimited storage
- no pro/prime features (atleast not yet)
- no ads
- official Telegram app is not heavy as Discord app (~50MP vs ~100MB)
- currently 2 bots running in the group, which keeps things automated
- almost 900 ppl in the group (best for discussion/user_feedback)
etc...

**Reply by VishalNehra** [reply]
Thanks for explaining. On top of that, all the bots running on discord (or anywhere else), need access and send all the messages for processing, so there's no way to be 100% sure your privacy specially on public groups that have automated bots. But the bots running on our telegram group are open source as well. So you may not use telegram as your go-to messaging app, but for our usecase it's pretty good.

**Reply by EmmanuelMess** [reply]
Please keep the conversation on topic, or create another discussion :).

**Comment by VishalNehra** [comment]
Amaze file utilities is now available on play store. 
@Jpersh27 
https://play.google.com/store/apps/details?id=com.amaze.fileutilities
